# Student Dataset Data Cleaning

## Objective
Prepare a student dataset for machine learning by cleaning data and selecting relevant features.

In [31]:
import pandas as pd
stud_data=pd.read_csv("student-dataset.csv")
stud_data.head()

id             name               nationality                 city  \
0   0        Kiana Lor                     China               Suzhou   
1   1   Joshua Lonaker  United States of America        Santa Clarita   
2   2    Dakota Blanco  United States of America              Oakland   
3   3  Natasha Yarusso  United States of America        Castro Valley   
4   4   Brooke Cazares                    Brazil  São José dos Campos   

   latitude  longitude gender  ethnic.group  age  english.grade  math.grade  \
0     31.31     120.62      F           NaN   22            3.5         3.7   
1     34.39    -118.54      M           NaN   22            2.9         3.2   
2     37.80    -122.27      F           NaN   22            3.9         3.8   
3     37.69    -122.09      F           NaN   20            3.3         2.8   
4    -23.18     -45.88      F           NaN   21            3.7         2.6   

   sciences.grade  language.grade  portfolio.rating  coverletter.rating  \
0             3.1             1.0                 4                 4.0   
1             3.6             5.0                 5                 4.0   
2             3.2             5.0                 3                 3.0   
3             3.2             5.0                 5                 2.0   
4             3.4             1.0                 4                 4.0   

   refletter.rating  
0                 4  
1                 5  
2                 4  
3                 4  
4                 5

In [5]:
stud_data.shape

(307, 16)

In [6]:
stud_data.columns

Index(['id', 'name', 'nationality', 'city', 'latitude', 'longitude', 'gender',
       'ethnic.group', 'age', 'english.grade', 'math.grade', 'sciences.grade',
       'language.grade', 'portfolio.rating', 'coverletter.rating',
       'refletter.rating'],
      dtype='object')

**Dataset description**
This student performance dataset contains data of students accross countries and their performance in each subjects


## Dataset Overview
- Rows: 307
- Columns: 16
- Data includes demographic and academic performance features

## Step 1: Check for Missing Values

In [11]:
stud_data.isna()

id   name  nationality   city  latitude  longitude  gender  \
0    False  False        False  False     False      False   False   
1    False  False        False  False     False      False   False   
2    False  False        False  False     False      False   False   
3    False  False        False  False     False      False   False   
4    False  False        False  False     False      False   False   
..     ...    ...          ...    ...       ...        ...     ...   
302  False  False        False  False     False      False   False   
303  False  False        False  False     False      False   False   
304  False  False        False  False     False      False   False   
305  False  False        False  False     False      False   False   
306  False  False        False  False     False      False   False   

     ethnic.group    age  english.grade  math.grade  sciences.grade  \
0            True  False          False       False           False   
1            True  False          False       False           False   
2            True  False          False       False           False   
3            True  False          False       False           False   
4            True  False          False       False           False   
..            ...    ...            ...         ...             ...   
302          True  False          False       False           False   
303          True  False          False       False           False   
304          True  False          False       False           False   
305          True  False          False       False           False   
306          True  False          False       False           False   

     language.grade  portfolio.rating  coverletter.rating  refletter.rating  
0             False             False               False             False  
1             False             False               False             False  
2             False             False               False             False  
3             False             False               False             False  
4             False             False               False             False  
..              ...               ...                 ...               ...  
302           False             False               False             False  
303           False             False               False             False  
304           False             False               False             False  
305           False             False               False             False  
306           False             False               False             False  

[307 rows x 16 columns]

In [32]:
stud_data= stud_data.drop(columns=['ethnic.group']

### Removing Irrelevant Columns
The column 'ethnic.group' was removed to avoid redundancy and potential bias,
and because it is not required for the current prediction task.


## Target Variable Selection
The target variable is `language.grade`, as the goal is to predict language
performance based on demographic and academic features.


In [45]:
X=stud_data[['nationality','city']]

In [46]:
y=stud_data['english.grade']

Step 3: Data Cleaning
Check for missing values in all columns
Handle missing values:
Numerical columns → fill with mean
Verify that no missing values remain
✍️ Explain why missing values must be handled before ML.
